compiler/rustc_incremental/src/persist/save.rs

```diff
@@ -139,6 +141,7 @@ fn encode_query_cache(tcx: TyCtxt<'_>, encoder: &mut FileEncoder) -> FileEncodeR
     tcx.sess.time("incr_comp_serialize_result_cache", || tcx.serialize_query_result_cache(encoder))
 }
 
+/// Builds the dependency graph.
```

Comment:
This function creates a file (the so-called staging dep-graph). When the dep-graph is modified by a query execution, the dependency info is not kept in memory but directly output to this file. `save_dep_graph` finalizes the staging dep-graph and moves it to the (permanent) dep-graph path.